src/commands/deploy/FunctionAppListStep.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type Site } from "@azure/arm-appservice";
+import { parseAzureResourceGroupId, uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { AzureWizardPromptStep, createSubscriptionContext, nonNullProp, type AzureWizardExecuteStep, type IAzureQuickPickItem, type IWizardOptions } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { projectLanguageSetting } from "../../constants";
+import { type ICreateFunctionAppContext } from "../../tree/SubscriptionTreeItem";
+import { createWebSiteClient } from "../../utils/azureClients";
+import { getWorkspaceSetting, getWorkspaceSettingFromAnyFolder } from "../../vsCodeConfig/settings";
+import { createCreateFunctionAppComponents } from "../createFunctionApp/createCreateFunctionAppComponents";
+import { type IFuncDeployContext } from "./deploy";
+
+export class FunctionAppListStep extends AzureWizardPromptStep<IFuncDeployContext> {
+    public async prompt(context: IFuncDeployContext): Promise<void> {
+        context.site = (await context.ui.showQuickPick(this.getPicks(context), { placeHolder: vscode.l10n.t("Select a function app") })).data;
+    }
+
+    public shouldPrompt(context: IFuncDeployContext): boolean {
+        return !context.site;
+    }
+
+    private async getPicks(context: IFuncDeployContext): Promise<IAzureQuickPickItem<Site | undefined>[]> {
+        const client = await createWebSiteClient([context, createSubscriptionContext(nonNullProp(context, 'subscription'))]);
+        const sites = (await uiUtils.listAllIterator(client.webApps.list()));
+        const qp: IAzureQuickPickItem<Site | undefined>[] = sites.filter(s => !!s.kind?.includes('functionapp')).map(fa => {
+            return {
+                label: nonNullProp(fa, 'name'),
+                description: parseAzureResourceGroupId(nonNullProp(fa, 'id')).resourceGroup,
+                data: fa
+            }
+        });
+
+        qp.unshift({ label: '$(plus) Create new function app', description: '', data: undefined });
+        return qp;
+    }
+
+    public async getSubWizard(context: IFuncDeployContext): Promise<IWizardOptions<IFuncDeployContext> | undefined> {
+        if (!context.site) {
```

Comment:
Maybe add a comment here. It seems like this the pivotal place where the code decides if we're creating a FA or not for deploy.

Reply:
I would also like to suggest using a guard clause here to clean this up a bit more :)

Reply:
Could you clarify what you mean by using a guard clause here? I'm not really sure what I would be type guarding in this case.

Reply:
Hey guy, I already did that.

Reply:
🫡